ethcore/types/src/errors/ethcore_error.rs

```diff
@@ -43,16 +55,19 @@ pub enum EthcoreError {
 	Io(ethcore_io::IoError),
 	/// Error concerning the Rust standard library's IO subsystem.
 	#[display(fmt = "Std Io error: {}", _0)]
-	StdIo(::std::io::Error),
+	StdIo(std::io::Error),
 	/// Error concerning TrieDBs.
 	#[display(fmt = "Trie error: {}", _0)]
 	Trie(TrieError),
 	/// Error concerning EVM code execution.
 	#[display(fmt = "Execution error: {}", _0)]
 	Execution(ExecutionError),
 	/// Error concerning block processing.
-	#[display(fmt = "Block error: {}", _0)]
+	#[display(fmt = "{}", _0)]
 	Block(BlockError),
+	/// TODO(niklasad1): document
+	#[display(fmt = "{}", _0)]
+	BadBlock(BlockErrorWithData),
```

Comment:
Now `Block` and `BadBlock` are redundant because both contains `BlockError`.
I have an idea to split cheap `Block processing` into a different variant e.g., `Header` then we could maybe just have `BadBlock/Block` and `Header/CheapBlockVerification`

Reply:
I will put it on ice, because it assumes that `CheapBlockVerification` will not report `block bytes` and I'm not sure if it is sufficient to report `BadHeader` in those scenarios probably but it is a mess to fix.